File: gianmartind/src/main/java/com/github/gianmartind/view/HomeFragment.java
package com.github.gianmartind.view;

import android.content.Context;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.github.gianmartind.R;
import com.github.gianmartind.FragmentListener;

public class HomeFragment extends Fragment implements View.OnClickListener {
    Button startGame, highScore, settings, exit;
    CustomToast toast;
    FragmentListener fragmentListener;
    MediaPlayer song;

    public HomeFragment(){}

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_home, container, false);
        View toastView = inflater.inflate(R.layout.custom_toast, container, false);
        this.toast = new CustomToast(toastView, getActivity().getApplicationContext());
        this.startGame = view.findViewById(R.id.btn_start);
        this.highScore = view.findViewById(R.id.btn_high_score);
        this.settings = view.findViewById(R.id.btn_settings);
        this.exit = view.findViewById(R.id.btn_exit);

        this.startGame.setOnClickListener(this);
        this.highScore.setOnClickListener(this);
        this.settings.setOnClickListener(this);
        this.exit.setOnClickListener(this);

        this.song = MediaPlayer.create(getActivity(),R.raw.canon);

        //this.song.setLooping(true);
        return view;
    }

    public void onAttach(Context context){
        super.onAttach(context);
        if(context instanceof FragmentListener){
            this.fragmentListener = (FragmentListener) context;
        } else{
            throw new ClassCastException(context.toString() + " must implement FragmentListener");
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        this.song.start();
    }

    @Override
    public void onPause() {
        super.onPause();
        this.song.stop();
    }

    @Override
    public void onClick(View v) {
        if(v == this.startGame){
            this.fragmentListener.changePage(1);
        } else if(v == this.highScore){
            this.fragmentListener.changePage(3);
        } else if(v == this.settings){
            this.fragmentListener.changePage(4);
        } else if(v == this.exit){
            this.fragmentListener.closeApplication();
        }
    }
}

File: obedkristiaji/src/main/java/com/example/pianotiles/view/HomeFragment.kt
package com.example.pianotiles.view

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.pianotiles.databinding.FragmentHomeBinding
import com.example.pianotiles.presenter.IMainPresenter


class HomeFragment : Fragment() {
    private lateinit var binding: FragmentHomeBinding
    private lateinit var listener: IMainActivity
    private lateinit var presenter: IMainPresenter

    init {

    }

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        this.binding = FragmentHomeBinding.inflate(inflater, container, false)

        this.init()

        this.binding.btnBack.setOnClickListener {
            this.listener.changePage(2)
        }

        this.binding.btnSetting.setOnClickListener {
            this.listener.showSetting()
        }

        this.binding.btnExit.setOnClickListener {
            this.listener.closeApplication()
        }

        return this.binding.root
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if(context is IMainActivity) {
            this.listener = context as IMainActivity
        } else {
            throw ClassCastException(context.toString()
                    + " must implement FragmentListener")
        }
    }

    companion object {
        fun newInstance(presenter: IMainPresenter): HomeFragment {
            val fragment: HomeFragment = HomeFragment()
            fragment.presenter = presenter
            return fragment
        }
    }

    fun setScore(score: Int) {
        this.binding.tvScore.setText("High Score: " + score.toString())
    }

    private fun init() {
        this.setScore(this.presenter.getHS())
        this.presenter.setStartHome(true)
    }
}